File: cpp/questions/generate.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        std::vector<std::vector<int>> ans;        
        ans.reserve(numRows);
        ans.emplace_back(1, 1);
        for (int i=1; i<numRows; ++i) {
            const auto& pr = ans[i-1];
            std::vector<int> cr(i+1, 1);
            for (int j=1; j<i; ++j) {
                cr[j] = pr[j-1] + pr[j];
            }
            ans.emplace_back(cr);
        }
        return ans;
    }
};

File: python/questions/generate.py
from typing import List

class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        ans = []
        ans.append([1])
        for i in range(1, numRows):
            pr = ans[i-1]
            cr = [1] * (i+1)
            for j in range(1, i):
                cr[j] = pr[j] + pr[j-1]
            ans.append(cr)
        return ans
